### PR TITLE
convert mscoffobj.c to D

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/mscoffobj.d [hunk 1]
```diff
@@ -1907,36 +1925,35 @@ void MsCoffObj_func_term(Symbol *sfunc)
  * Output a public definition.
  * Input:
  *      seg =           segment index that symbol is defined in
- *      s ->            symbol
+ *      s .            symbol
```
Comment (rainers) [inline src/dmd/backend/mscoffobj.d]: I guess DAutoTest doesn't like this one.

File: src/dmd/backend/mscoffobj.d [hunk 2]
```diff
@@ -1976,17 +1993,17 @@ int MsCoffObj_external_def(const char *name)
...
+    symbuf.write(&s, (s).sizeof);
+    s.Sxtrnnum = 1;
     return 1;
 }
 
 /*******************************
  * Output a common block definition.
  * Input:
- *      p ->    external identifier
+ *      p .    external identifier
```
Comment (rainers) [inline src/dmd/backend/mscoffobj.d]: and this.

File: src/dmd/backend/mscoffobj.d [hunk 3]
```diff
@@ -2250,27 +2263,27 @@ void MsCoffObj_reftocodeseg(segidx_t seg,targ_size_t offset,targ_size_t val)
...
+        buf.setsize(save);
 }
 
 /*******************************
  * Refer to an identifier.
  * Input:
  *      seg =   where the address is going (CODE or DATA)
  *      offset =        offset within seg
- *      s ->            Symbol table entry for identifier
+ *      s .            Symbol table entry for identifier
```
Comment (rainers) [inline src/dmd/backend/mscoffobj.d]: and this

**Comment by WalterBright** [comment]
@rainers thanks for finding the problem. I don't know how you found it in the log, all I could find was:
```
The dependency resolution process is taking too long. The dependency graph is likely hitting a pathological case in the resolution algorithm. Please file a bug report at https://github.com/dlang/dub/issues and mention the package recipe that reproduces this error.
posix.mak:766: recipe for target 'dpl-docs' failed
```
so I assumed it was some environmental problem.

**Comment by rainers** [comment]
> @rainers thanks for finding the problem. I don't know how you found it in the log

I just noticed during review, I think there will be more (IIRC some parameter names in the doc didn't match the function signature, but couldn't find them now). DAutoTest seems to fail for other reasons on all PRs, see e.g. https://github.com/dlang/dmd/pull/8878#issuecomment-433679318

File: src/win32.mak [hunk 4]
```diff
@@ -578,8 +578,8 @@ $G/gsroa.obj : $C\gsroa.d
 $G/md5.obj : $C\md5.d
 	$(HOST_DC) -c -of$@ $(DFLAGS) -betterC -mv=dmd.backend=$C $C\md5
 
-$G/mscoffobj.obj : $C\mscoff.h $C\mscoffobj.c
-	$(CC) -c -o$@ $(MFLAGS) -I$D;$(ROOT) -I$G $C\mscoffobj
+$G/mscoffobj.obj : $C\mscoff.d $C\mscoffobj.d
```
Comment (rainers) [inline src/win32.mak]: This compiles the 2 files into different object files, but not both are passed to the linker. Simplest to compile them seperately.

Comment (WalterBright) [reply src/win32.mak]: One wonders why that suddenly stopped working (and it works on my machine). Sigh.

**Review by thewilsonator — COMMENTED** [review]
I know you try to not make unnecessary changes when converting files but it's probably a good idea to address this even if you just comment the test and one of the branches.

File: src/dmd/backend/mscoffobj.d [hunk 5]
```diff
@@ -757,209 +765,211 @@ void MsCoffObj_term(const char *objfilename)
...
-                                rel.r_symndx = s->Sxtrnnum;
+                                rel.r_vaddr = cast(uint)r.offset;
+                                rel.r_symndx = s.Sxtrnnum;
                             }
                             else
                             {
-                                rel.r_vaddr = r->offset;
-                                rel.r_symndx = s->Sxtrnnum;
+                                rel.r_vaddr = cast(uint)r.offset;
+                                rel.r_symndx = s.Sxtrnnum;
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/backend/mscoffobj.d]: I commented them out.

**Comment by wilzbach** [comment]
Rebased to restart DAutoTest

**Review by andralex — APPROVED** [review]
onward!

**Pull request merged**